src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -232,43 +481,110 @@ public Authentication authenticate(Authentication authentication) throws Authent
+                        for (String userGroup : usersGroups) {                    	
+                        	// copy any of the user's permissions from the prior auth mgr to our new one
+                        	for (PermissionGroup pg : permissionGroups) {
+                        		for (Permission p : pg.getPermissions()) {
+                        			if (existingAuthMgr.hasPermission(userGroup, p)) {
+                        				newAuthMgr.add(p, userGroup);
+                        			}
+                        		}
+                        	}
+                        }
+                        
+                    	if (addInsteadOfCopy) {
+                        	if (LOGGER.isLoggable(Level.FINE))
+                        		LOGGER.fine(String.format("onSuccess: adding permissions to new user %s based on openshift roles %s", info.getName(), allowedRoles));
+                        	
+                        	// map OpenShift user based on role to Jenkins user with analogous permissions
+                        	//TODO can we minimally assume view role ?  assuming so while waiting for the build.oenshift.io.jenkins related changes we think are coming
+                        	// and see if a self-sar against the view role comes back and says allowed (currently returns allowed as false for the view role)
+                        	newAuthMgr.add(Hudson.READ, info.getName());
+                        	newAuthMgr.add(Item.READ, info.getName());
+                			if (allowedRoles.contains("edit") || allowedRoles.contains("admin")) {
+                				newAuthMgr.add(Item.BUILD, info.getName());
+                				newAuthMgr.add(Item.CONFIGURE, info.getName());
+                				newAuthMgr.add(Item.CREATE, info.getName());
+                				newAuthMgr.add(Item.DELETE, info.getName());
+                				newAuthMgr.add(Item.WORKSPACE, info.getName());
+                				newAuthMgr.add(SCM.TAG, info.getName());
+                			}
+                			if (allowedRoles.contains("admin")) {
+                				newAuthMgr.add(Computer.CONFIGURE, info.getName());
+                				newAuthMgr.add(Computer.DELETE, info.getName());
+                				newAuthMgr.add(Hudson.ADMINISTER, info.getName());
+                				newAuthMgr.add(Hudson.READ, info.getName());
+                				newAuthMgr.add(Run.DELETE, info.getName());
+                				newAuthMgr.add(Run.UPDATE, info.getName());
+                				newAuthMgr.add(View.CONFIGURE, info.getName());
+                				newAuthMgr.add(View.CREATE, info.getName());
+                				newAuthMgr.add(View.DELETE, info.getName());
+                			}
+                    	}
+            			
+            			Jenkins.getInstance().setAuthorizationStrategy(newAuthMgr);
```

Comment:
I deleted the previous brainstorm on http session related events ... refreshed my mem with the java doc.  None of the 3 listeners are called on each session access.  The callouts present won't serve our periodic polling needs.

A servlet filter would work, but you specify those in the web.xml.  I see some indirect refs to tweaking their web.xml in stackoverflow, but nothing that gives me confidence it is something we should be doing.

If refresh in between sessions is not sufficient, we are back to polling in a background thread, or polling say as part of the pipeline plugin's operation.  There are standard Jenkins api that would allow it to get the openshift oauth login plugin.